### PR TITLE
[MIN-25] Add step to deploy model

### PR DESCRIPTION
**Pull request by osw282**
This PR is still WIP and implements a model deployment step to deploy a base flan model we obtain from the `fetch_mdoel_step.py`. This step is update so that we can return the model and tokenizer artifact uri to the deployer step. 4 scripts are implemented under the `deploy_model_step` folder in order to deploy the flan LLM using seldon. The deployment pipeline can now be run successfully without cloud resources hosted with `matcha`.

Below are the ZenML stack set up required to run the deployment pipeline, you will have to export each of the variable below based on the output from `matcha`:
```
zenml integration install huggingface pytorch azure kubernetes seldon -y

zenml init

zenml disconnect

az acr login --name="$acr_registry_name"

zenml secret create az_secret --connection_string="$zenml_connection_string"
zenml container-registry register acr_registry -f azure --uri="$acr_registry_uri"
zenml artifact-store register az_store -f azure --path="$zenml_storage_path" --authentication_secret=az_secret
zenml image-builder register docker_builder --flavor=local
zenml model-deployer register seldon_deployer --flavor=seldon \
        --kubernetes_context=$k8s_context \
        --kubernetes_namespace=$seldon_workload_namespace \
        --base_url=http://$seldon_ingress_host
zenml stack register llm_test_stack -i docker_builder -a az_store -o default -c acr_registry --model_deployer=seldon_deployer --set

python run.py -d
```

To test the deploy model, we can curl it with:
```
curl -XPOST -H 'Content-Type: application/json' -d '{"data": {"ndarray": [{"text": "Hi can you summarize this!"}]}}' <endpoint_url>
```

TODO:
- Update the custom predict function, the key of the payload could be "question" instead of "text".
- Implements tests for new functions and steps implemented.
- Possibly implement a set up script like the LLM example from Shubham so that ZenML stack can be set up automatically.

If the endpoint is not reachable for some reason, we can debug it by looking into the logs of the pod of the `classifier container` assuming `init container` is completed:
`kubectl logs <pod_name> -n matcha-seldon-workloads -c classifier`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,15 +23,71 @@ Mental health problems are something that everyone struggles with at various poi
...
 The repository for this project is one method where you can monitor progress - we're always raising pull requests to move the project forward. Another way you can do this is to follow our blog. We'll be posting periodic updates as we tick off sections of the project. If you're new to the project, the best place to start is our [project introduction](https://www.fuzzylabs.ai/blog-post/mindgpt-an-introduction).
 
 # &#127939; How do I get started?
 
-We're working on this! Coming soon...
+## Deployment pipeline
+
+To run the zenml deployment pipeline, the resources required by seldon have to provisioned. `matcha` tool can help you in provisioning these resources.
+
+Provision resources using `matcha`
```
Comment (JonoCX) [inline README.md]: There's a missing step here for installing matcha

File: README.md [hunk 2]
```diff
@@ -23,15 +23,71 @@ Mental health problems are something that everyone struggles with at various poi
...
+
+```bash
+matcha provision
+```
+
+Assign output for resources provisioned by `matcha` to variables.
+
+```bash
+function get_state_value() {
```
Comment (JonoCX) [inline README.md]: I'd probably put this into a bash script, e.g., `setup.sh` with any prerequisite steps added to the readme before telling the user to run the setup script

File: README.md [hunk 3]
```diff
@@ -23,15 +23,71 @@ Mental health problems are something that everyone struggles with at various poi
...
+k8s_context=$(get_state_value orchestrator k8s-context)
+acr_registry_uri=$(get_state_value container-registry registry-url)
+acr_registry_name=$(get_state_value container-registry registry-name)
+seldon_workload_namespace=$(get_state_value model-deployer workloads-namespace)
+seldon_ingress_host=$(get_state_value model-deployer base-url)
+```
+
+Setup zenml stack for deployment
+
+```bash
```
Comment (JonoCX) [inline README.md]: Same with this - keeps it a bit tidier (should have said on the call, apologies)

**Review by JonoCX — CHANGES_REQUESTED** [review]
A couple of changes to the readme and there aren't any tests, suppose the blocker is it being remote?

File: README.md (hunk 1 above)
Comment (dudeperf3ct) [reply README.md]: This is not battle tested. ie. we are facing some issues with installing matcha inside mindgpt. Still, I will include it.

**Review by JonoCX — APPROVED** [review]
LGTM
